packages/core/test/actions.test.ts

```diff
@@ -1142,7 +1203,10 @@ describe('entry/exit actions', () => {
       expect(eventReceived).toBe(true);
     });
 
-    it('sent events from exit handlers of a done child should be received by its children ', () => {
+    // TODO: Determine what is the correct behavior here
+    // We're incorrectly assuming that sending from child -> grandchild is synchronous
+    // A race condition can occur where the grandchild is stopped before an event is received
```

Comment:
This tests a very specific situation - with no other machines in the picture etc so a grandchild can't be just stopped for any other reason than because its parent here (`child`) gets stopped. 

No race condition is possible here even if we assume that sending events is asynchronous because this test tests that events sent from `child` to `grandchild` are sent "in order". It also tests that a sent event is sent **before** an implicit event that informs the `grandchild` to stop. In other words, it tests that we can send events from a machine to its children before notifying them about them being stopped and it tests that they can process that event (because the stop notification has not been delivered to them yet, or because it stays queued after that other event).